### PR TITLE
Pcap conditional v2.2.7

### PR DESCRIPTION
**Pull request by scottfgjordan**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/120

Describe changes:
- Addressed feedback from @jlucovsky in https://github.com/OISF/suricata/pull/6389
- Corrected a commit error from rebase so commit check should pass now.
- Rebased with latest master changes.

suricata-verify-pr: 542

### CODE REVIEW

File: src/stream-tcp-reassemble.c [hunk 1]
```diff
@@ -250,19 +262,64 @@ static void *TcpSegmentPoolAlloc(void)
...
         return NULL;
+
+    if (IsTcpSessionDumpingEnabled()) {
+        uint32_t memuse =
+                sizeof(TcpSegmentPcapHdrStorage) + sizeof(char) * TCPSEG_PKT_HDR_DEFAULT_SIZE;
+        if (StreamTcpReassembleCheckMemcap(sizeof(TcpSegment) + memuse) == 0) {
+            SCFree(seg);
+            return NULL;
+        }
+
```
Comment (jlucovsky) [inline src/stream-tcp-reassemble.c:273]: Missing call to `StreamTcpReassembleIncrMemuse` to adjust the memory used that's tracked by the memcap value.

File: src/util-datalink.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        g_datalink_value = datalink;
+    }
+}
+
+int DatalinkGetGlobalType(void)
+{
+    return g_datalink_value;
+}
+
+int DatalinkHasMultipleValues(void)
```
Comment (jlucovsky) [inline src/util-datalink.c:42]: Strongly suggest type `bool` for this

File: src/stream-tcp-reassemble.c (hunk 1 above)
Comment (scottfgjordan) [reply src/stream-tcp-reassemble.c:273]: Good catch! Thanks.

**Comment by scottfgjordan** [comment]
Replaced by: https://github.com/OISF/suricata/pull/6431

**Pull request closed without merging**